src/System.Runtime.Extensions/src/System/Environment.WinRT.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Windows.Foundation.Metadata;
+using Windows.Storage;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOption option)
+        {
+            // For testing we'll fall back if the needed APIs aren't present
+            if (ApiInformation.IsTypePresent("Windows.Storage.UserDataPaths"))
+            {
+                return GetFolderPathCoreCurrent(folder, option);
+            }
+            else
+            {
+                return GetFolderPathCoreFallBack(folder, option);
+            }
+
+        }
+
+        private static string GetFolderPathCoreCurrent(SpecialFolder folder, SpecialFolderOption option)
```

Comment:
> I'll circle back with more details

I see you merged but didn't see the results of this... what did you discover that made you decide not to address this?